feeds/wlan-ap/opensync/src/platform/openwrt/src/netifd/src/inet_conf.c

```diff
@@ -6,30 +6,38 @@
 
 #include "inet_conf.h"
 
-struct netifd_iface* netifd_add_inet_conf(struct schema_Wifi_Inet_Config *iconf)
-{
+static int vlan_count = 0;
 
+void netifd_add_inet_conf(struct schema_Wifi_Inet_Config *iconf)
+{
 	struct netifd_iface *piface = NULL;
 
+	if (strcmp(iconf->if_type, "bridge") && strcmp(iconf->if_type, "vlan")) {
+		return;
+	}
+
 	piface = netifd_iface_get_by_name(iconf->if_name);
 	if (piface == NULL)
 	{
 		piface = netifd_iface_new(iconf->if_name, iconf->if_type);
 		if (piface == NULL)
 		{
 			LOG(ERR, "netifd_add_inet_conf: %s: Unable to create interface.", iconf->if_name);
-			return NULL;
+			return;
 		}
 
-		if (!strcmp(iconf->if_type, "bridge") || !strcmp(iconf->if_type, "vlan"))
-		{
-			LOGN("Setting up dhsnif for %s", piface->if_base->inet.in_ifname);
+		if (!strcmp(iconf->if_name, "wan") || !strcmp(iconf->if_name, "lan")) {
 			netifd_inet_config_set(piface);
 			netifd_inet_config_apply(piface);
+		} else if (iconf->vlan_id_exists && iconf->vlan_id > 2) {
+			if (vlan_count < DHCP_SNIFF_MAX_VLAN && !strstr(iconf->if_name,"lan_")) {
+				netifd_inet_config_set(piface);
+				netifd_inet_config_apply(piface);
+				vlan_count++;
```

Comment:
I don't see anything that decrements this count.  If you delete you config and add a new one, does the DHCP sniff get started?

Reply:
Fixed with the new patch. Tested deletion and addition of a new config, Sniffing is always enabled on a max of 4 vlans.